File: kwkim526/큐,스택/3-올바른 괄호.cpp
#include<string>
#include <iostream>
#include <stack>

using namespace std;

bool solution(string s)
{
    stack<bool> isOpened;
    
    for(const auto& c : s)
    {
        if(c == '(')
        {
            isOpened.push(true);
        }
        else
        {
            if(isOpened.empty())
            {
                return false;
            }
            else
            {
                isOpened.pop();
            }
        }
    }
    
    return isOpened.empty();
}

File: hide-squid/스택-큐/3.올바른괄호.py
def solution(s):
    list_ = []
    
    if s[0] == ')':
        return False
    for parentheses in s:
        if parentheses == '(':
            list_.append(parentheses)
        else:
            if len(list_) == 0:
                return False
            list_.pop()
    if len(list_) == 0:
        return True
    else:
        return False